File: Stack_Pro_Lv2_뒤에 있는 큰 수 찾기.kt
import java.util.*

class Solution {
    fun solution(numbers: IntArray): IntArray {
        var answer: IntArray = IntArray(numbers.size) { -1 }
        val stack = Stack<Pair<Int, Int>>()
        for(i in numbers.indices) {
            while(stack.isNotEmpty() && stack.peek().second < numbers[i]) {
                answer[stack.pop().first] = numbers[i]
            }
            stack.add(Pair(i, numbers[i]))
        }
        return answer
    }
}

File: Stack_Pro_Lv2_뒤에 있는 큰 수 찾기.py
def solution(numbers):
    answer = [-1 for i in range(len(numbers))]
    stack = []
    for i in range(len(numbers)):
        while len(stack) > 0 and stack[-1][1] < numbers[i]:
            answer[stack.pop()[0]] = numbers[i]
        stack.append((i, numbers[i]))
    return answer
